coordinator/persistence-cassandra-4.1/src/main/java/io/stargate/db/cassandra/impl/Cassandra41Persistence.java

```diff
@@ -0,0 +1,594 @@
+import org.apache.cassandra.service.StorageService;
+import org.apache.cassandra.stargate.exceptions.PersistenceException;
+import org.apache.cassandra.stargate.transport.ProtocolVersion;
+import org.apache.cassandra.transport.Message;
+import org.apache.cassandra.transport.Message.Request;
+import org.apache.cassandra.transport.messages.BatchMessage;
+import org.apache.cassandra.transport.messages.ErrorMessage;
+import org.apache.cassandra.transport.messages.ExecuteMessage;
+import org.apache.cassandra.transport.messages.PrepareMessage;
+import org.apache.cassandra.transport.messages.QueryMessage;
+import org.apache.cassandra.transport.messages.ResultMessage;
+import org.apache.cassandra.transport.messages.StartupMessage;
+import org.apache.cassandra.utils.ByteBufferUtil;
+import org.apache.cassandra.utils.FBUtilities;
+import org.apache.cassandra.utils.JVMStabilityInspector;
+import org.apache.cassandra.utils.MD5Digest;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class Cassandra41Persistence
+    extends AbstractCassandraPersistence<
+        Config,
+        KeyspaceMetadata,
+        TableMetadata,
+        ColumnMetadata,
+        UserType,
+        IndexMetadata,
+        ViewMetadata> {
+  private static final Logger logger = LoggerFactory.getLogger(Cassandra41Persistence.class);
+
+  private static final boolean USE_TRANSITIONAL_AUTH =
+      Boolean.getBoolean("stargate.cql_use_transitional_auth");
+
+  /*
+   * Initial schema migration can take greater than 2 * MigrationManager.MIGRATION_DELAY_IN_MS if a
+   * live token owner doesn't become live within MigrationManager.MIGRATION_DELAY_IN_MS.
+   */
+  private static final int STARTUP_DELAY_MS =
+      Integer.getInteger(
+          "stargate.startup_delay_ms",
+          3 * 60000); // MigrationManager.MIGRATION_DELAY_IN_MS is private
+
+  // SCHEMA_SYNC_GRACE_PERIOD should be longer than MigrationManager.MIGRATION_DELAY_IN_MS to allow
+  // the schema pull tasks to be initiated, plus some time for executing the pull requests plus
+  // some time to merge the responses. By default the pull task timeout is equal to
+  // DatabaseDescriptor.getRpcTimeout() (10 sec) and there are no retries. We assume that the merge
+  // operation should complete within the default MIGRATION_DELAY_IN_MS.
+  private static final Duration SCHEMA_SYNC_GRACE_PERIOD =
+      Duration.ofMillis(Long.getLong("stargate.schema_sync_grace_period_ms", 2 * 60_000 + 10_000));
+
+  private final SchemaCheck schemaCheck = new SchemaCheck();
+
+  private LocalAwareExecutorPlus executor;
+
+  private CassandraDaemon daemon;
+  private Authenticator authenticator;
+  private QueryInterceptor interceptor;
+
+  // C* listener that ensures that our Stargate schema remains up-to-date with the internal C* one.
+  private SchemaChangeListener schemaChangeListener;
+  private AtomicReference<AuthorizationService> authorizationService;
+
+  public Cassandra41Persistence() {
+    super("Apache Cassandra");
```

Comment:
Would be great to know what this is used for (DSE seems to use "DataStax Enterprise") -- probably not safe to change if drivers use it to determine logic etc. But otherwise would be nice to use something like "Apache Cassandra 4.1.x (Stargate)" (not suggesting we do, something would probably break)